lightning/src/ln/channelmanager.rs

```diff
@@ -1836,22 +1889,18 @@ impl<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelManager<M, T, K, F
 		let result: Result<(), _> = loop {
 			let mut channel_state_lock = self.channel_state.lock().unwrap();
 			let channel_state = &mut *channel_state_lock;
-			match channel_state.by_id.entry(channel_id.clone()) {
+			let per_peer_state = self.per_peer_state.read().unwrap();
+
+			if let None = per_peer_state.get(counterparty_node_id) {
+				return Err(APIError::APIMisuseError { err: format!("Can't find a peer matching the passed counterparty node_id {}", counterparty_node_id) });
```

Comment:
Let me know if you think this message + the message when the channel can't be found is ok (commit 030ef23010401180fbfa55012d4fdca8df91170c), or if you prefer some other wording.